src/FeedsCleanup/Program.cs

```diff
@@ -0,0 +1,199 @@
+                                             .Where(p => !matchedVersion.Contains(Path.GetFileName(p))));
+                }
+                else
+                {
+                    deleteCandiates.Add(latestPackageFolder);
+                    Warn("Project {0} doesn't exist in this Coherence build hence it will be deleted from {1}", projectName, ciLatestPackagesFolder);
+                }
+            }
+
+            foreach (var candiate in deleteCandiates)
+            {
+                Info("Delete folder {0}", candiate);
+
+                if (!whatif)
+                {
+                    Info("Actual delete!");
+                    Directory.Delete(candiate, recursive: true);
+                }
+            }
+
+            return true;
+        }
+
+        private static bool RemoveFromdMyGet(IEnumerable<string> coherentPackageNames,
+                                             string feed,
+                                             string apiKey,
+                                             bool whatif)
+        {
+            Console.WriteLine("Listing packages on {0}", feed);
+            var task = Task.Run(() =>
+            {
+                var repo = PackageRepositoryFactory.Default.CreateRepository(feed);
+                return repo.GetPackages().ToList();
+            });
+
+            var remotePackagesIds = task.Result.Select(package => package.Id).Distinct();
+            var localPackagsFullNames = coherentPackageNames.Select(n => Path.GetFileNameWithoutExtension(n));
+
+            var deleteCandidateIds = new HashSet<string>();
+            foreach (var id in remotePackagesIds)
+            {
+                if (!localPackagsFullNames.Any(fullname => fullname.Contains(id)))
+                {
+                    deleteCandidateIds.Add(id);
+                }
+            }
+
+            var requests = task.Result.Where(package => deleteCandidateIds.Contains(package.Id))
+                                      .Select(package => string.Format("{0}/package/{1}/{2}?hardDelete=true", feed.TrimEnd('/'), package.Id, package.Version));
+
+            foreach (var r in requests)
+            {
+                Console.WriteLine("DELETE {0}", r);
+            }
+
+            if (!whatif)
+            {
+                Console.WriteLine("Sending delete requests");
+
+                var httpClient = new System.Net.Http.HttpClient();
+                httpClient.DefaultRequestHeaders.Add("X-NuGet-ApiKey", apiKey);
+
+                var tasks = requests.Select(request => httpClient.DeleteAsync(request));
+                Task.WaitAll(task);
+
+                foreach (var failure in tasks.Where(t => !t.Result.IsSuccessStatusCode))
+                {
+                    Error("Failed request {0}", failure.Result.RequestMessage.RequestUri.AbsolutePath);
+                }
+            }
+
+            return true;
+        }
+
+        private static void Error(string message, params string[] parameters)
```

Comment:
Use TeamCity's error logging format if this is meant to be run on the server - https://github.com/aspnet/Coherence/blob/dev/src/SanityCheck/Program.cs#L294-L306